### ISSUE TITLE
Remove connexion version <= 2.7.0 limit after working around mock issue with newer connexion

### ISSUE THREAD

**Issue by AmandaBirmingham** [biocore/microsetta-private-api#369]
As of 07/22/21:
Beware!  Connexion has released two new versions in the last 14 days (2.8.0 and 2.9.0) and both are incompatible with the way our microsetta-private-api unit tests (e.g. in test_admin_api.py, test_api.py, and test_integration.py) mock get and post requests.   It appears that as of connexion 2.8.0, the connexion authors have introduced a new object called a security handler factory that now handles requests, and it passes down the chain an argument called “required_scopes”.  That argument appears to be propagated all the way down to python’s “mock_call” method, in which a call to “mock_verify” fails because it is receiving an unexpected parameter called “required_scopes”. (edited) 

This causes every test in the entire microsetta-private-api test suite that mocks a get or post request to fail :man-facepalming::skin-tone-4:.  Changing the pip requirements to prevent using a connexion version later than 2.7.0 makes everything pass again.

Dunno if this is a long-term solution, though … but figuring out how to change the mocking to avoid this is beyond me …

**Comment by wasade** [comment]
Thanks, @AmandaBirmingham. I believe this is solved with pinning the [connexion version](https://github.com/biocore/microsetta-private-api/blob/master/ci/pip_requirements.txt#L2)?

**Comment by AmandaBirmingham** [comment]
@wasade yes, changing the pip requirements to prevent using a connexion version later than 2.7.0 makes everything pass again ... but presumably eventually being stuck on an old version of connection will itself become a problem, so I'm just recording this for someday

**Comment by wasade** [comment]
Makes sense, thanks!
